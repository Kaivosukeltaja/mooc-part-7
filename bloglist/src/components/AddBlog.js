import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { addBlog } from '../actions/blogs'
import { showNotification } from '../actions/notifications'

const initialState = {
  title: '',
  author: '',
  url: '',
}

class AddBlog extends Component {
  constructor(props) {
    super(props)
    this.state = initialState
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value })    
  }

  async handleSubmit(event) {
    event.preventDefault()
    this.props.addBlog({
      title: this.state.title, 
      author: this.state.author, 
      url: this.state.url
    })
    this.props.showNotification(`Added new blog ${this.state.title}`)
    this.setState(initialState)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h2>Create new blog</h2>
        Title <input name="title" value={this.state.title} onChange={this.handleChange} /><br />
        Author <input name="author" value={this.state.author} onChange={this.handleChange} /><br />
        URL <input name="url" value={this.state.url} onChange={this.handleChange} /><br />
        <input type="submit" value="Create blog" />
      </form>
    )
  }
}

AddBlog.propTypes = {
  addBlog: PropTypes.func.isRequired,
  showNotification: PropTypes.func.isRequired,
}

const mapDispatchToProps = {
  addBlog,
  showNotification,
}

export default connect(null, mapDispatchToProps)(AddBlog)
