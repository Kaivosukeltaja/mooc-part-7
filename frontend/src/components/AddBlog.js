import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { addBlog } from '../actions/blogs'
import { showNotification } from '../actions/notifications'

const styles = {
  input: {
    display: 'block',
    margin: '0.25em 0',
    padding: '1em',
    maxWidth: '90%',
  },
  createButton: {
    backgroundColor: '#229922',
    color: '#ffffff',
    padding: '1em',
    margin: '0.25em 0',
    borderWidth: 0,
    borderRadius: '4px',
    fontWeight: 'bold',
  }
}

const initialState = {
  title: '',
  author: '',
  url: '',
}

export class AddBlog extends Component {
  constructor(props) {
    super(props)
    this.state = initialState
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value })    
  }

  async handleSubmit(event) {
    event.preventDefault()
    this.props.addBlog({
      title: this.state.title, 
      author: this.state.author, 
      url: this.state.url
    })
    this.props.showNotification(`Added new blog ${this.state.title}`)
    this.setState(initialState)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <h2>Create new blog</h2>
        <input placeholder="Title" name="title" style={styles.input} value={this.state.title} onChange={this.handleChange} /><br />
        <input placeholder="Author" name="author" style={styles.input} value={this.state.author} onChange={this.handleChange} /><br />
        <input placeholder="URL" name="url" style={styles.input} value={this.state.url} onChange={this.handleChange} /><br />
        <input type="submit" value="Create blog" style={styles.createButton} />
      </form>
    )
  }
}

AddBlog.propTypes = {
  addBlog: PropTypes.func.isRequired,
  showNotification: PropTypes.func.isRequired,
}

const mapDispatchToProps = {
  addBlog,
  showNotification,
}

export default connect(null, mapDispatchToProps)(AddBlog)
