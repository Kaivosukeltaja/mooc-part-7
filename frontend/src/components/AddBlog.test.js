import React from 'react'
import renderer from 'react-test-renderer'
import { AddBlog } from './AddBlog'

describe('<AddBlog />', () => {
  const addBlog = jest.fn()
  const showNotification = jest.fn()

  it('renders correctly', () => {
    const tree = renderer
      .create(<AddBlog addBlog={addBlog} showNotification={showNotification} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
