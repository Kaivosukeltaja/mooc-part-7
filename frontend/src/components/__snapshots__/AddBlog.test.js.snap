// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBlog /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Create new blog
  </h2>
  <input
    name="title"
    onChange={[Function]}
    placeholder="Title"
    style={
      Object {
        "display": "block",
        "margin": "0.25em 0",
        "maxWidth": "90%",
        "padding": "1em",
      }
    }
    value=""
  />
  <br />
  <input
    name="author"
    onChange={[Function]}
    placeholder="Author"
    style={
      Object {
        "display": "block",
        "margin": "0.25em 0",
        "maxWidth": "90%",
        "padding": "1em",
      }
    }
    value=""
  />
  <br />
  <input
    name="url"
    onChange={[Function]}
    placeholder="URL"
    style={
      Object {
        "display": "block",
        "margin": "0.25em 0",
        "maxWidth": "90%",
        "padding": "1em",
      }
    }
    value=""
  />
  <br />
  <input
    style={
      Object {
        "backgroundColor": "#229922",
        "borderRadius": "4px",
        "borderWidth": 0,
        "color": "#ffffff",
        "fontWeight": "bold",
        "margin": "0.25em 0",
        "padding": "1em",
      }
    }
    type="submit"
    value="Create blog"
  />
</form>
`;
